src/commands/view/summary.ts

```diff
@@ -1,102 +1,154 @@
+ */
+interface CommandFlags {
+  input: string[];
+  output?: string;
+  format: string;
+  stdout: boolean;
+  'print-pretty': boolean;
+  'title-table'?: boolean;
+  logLevel?: string;
+  help: string | undefined;
+}
+
+/**
+ * Summary Class
+ *
+ * This class represents a command in the CLI that provides a quick compliance overview of an HDF file.
+ * It includes methods to convert the data to different formats (JSON, YAML, Markdown) and to print the data to the console or write it to a file.
+ *
+ * @class
+ * @public
+ * @property {string[]} aliases - The aliases for this command. Users can invoke this command by typing 'summary'.
+ */
 export default class Summary extends Command {
+  /**
+   * @property {string[]} aliases - Alternative command name.
+   */
   static aliases = ['summary']
 
-  static usage = 'view summary -i <hdf-file> [-h] [-j] [-o <output>]'
+  /**
+   * @property {ReturnType<typeof createWinstonLogger>} logger - Winston logger for this command.
+   */
+  private logger: ReturnType<typeof createWinstonLogger> = createWinstonLogger('View Summary:');
 
-  static description = 'Get a quick compliance overview of an HDF file '
+  /**
+   * @property {string} description - Command description displayed in the help message.
+   */
+  static description = 'Generate a comprehensive summary of compliance data, including totals and counts, from your HDF files.\n The output can be displayed in the console, or exported as YAML, JSON, or a GitHub-flavored Markdown table.';
 
   static flags = {
-    help: Flags.help({char: 'h'}),
-    input: Flags.string({char: 'i', required: true, multiple: true, description: 'Input HDF files'}),
-    json: Flags.boolean({char: 'j', required: false, description: 'Output results as JSON'}),
-    output: Flags.string({char: 'o', required: false}),
-  }
+    input: Flags.string({char: 'i', required: true, multiple: true, description: 'Specify input HDF file(s)', helpGroup: IO_GROUP}),
+    output: Flags.string({char: 'o', description: 'Specify output file(s)', helpGroup: IO_GROUP}),
+    format: Flags.string({char: 'f', description: 'Specify output format', helpGroup: FORMATTING_GROUP, options: FORMAT_OPTIONS, default: 'yaml'}),
+    stdout: Flags.boolean({char: 's', description: 'Enable printing to console', default: true, allowNo: true, helpGroup: IO_GROUP}),
+    'print-pretty': Flags.boolean({char: 'r', description: 'Enable human-readable data output', helpGroup: FORMATTING_GROUP, default: true, allowNo: true}),
+    'title-table': Flags.boolean({char: 't', description: 'Add titles to the markdown table(s)', helpGroup: FORMATTING_GROUP, default: true, allowNo: true}),
+    logLevel: Flags.string({char: 'l', description: 'Set log level', helpGroup: 'debugging', default: DEFAULT_LOG_LEVEL}),
+    help: Flags.help({char: 'h', description: 'Show help information'}),
```

Comment:
```suggestion
    help: Flags.help({char: 'h', description: 'Show help information', helpGroup: 'help'}),
```

Consider adding this so the help output doesn't just say "FLAGS" and require the user to see that there are more types of flags underneath.

Reply:
Plese go ahead with that change and we can see if we want to standardize that across the commands, I think in the end we will as part of the general cleanup of user doc